CONTRIBUTING.md

```diff
@@ -209,6 +209,52 @@ for (const blockResult of blocksResult) {
 }
 ```
 
+## Managing and Opening Feature/Large PRs
+
+To maintain code quality, improve collaboration, and ensure clarity in large or complex changes, we follow these guidelines when opening pull requests (PRs). Depending on the nature of the change, PRs fall into three categories:
+
+### 1. Single, Complete PR
+
+If the PR contains a self-contained and complete feature or bug fix that does not require major refactoring or cross-team discussions, then:
+
+- Clearly explain the rationale and motivation behind the change in the PR description.
+- Provide relevant context, including:
+- Problem the PR is solving.
+- Why the approach was chosen.
+- Any alternatives considered.
+- If the PR modifies critical code paths, add references to relevant issues, benchmarks, or related discussions.
+- Ensure the PR adheres to our standard PR etiquette and commit message guidelines.
+
+### 2. PR with Major Refactoring
+
+If the PR involves significant code refactoring, structural changes, or fundamental modifications where team input is needed:
+
+- Start a GitHub Discussion before writing code.
+- Outline the problem, your proposed approach, and any alternative solutions.
+- Request feedback and build consensus with the team.
+- Link the discussion in the PR description once consensus is reached.
+- If changes affect multiple packages or require coordination with ongoing development, summarize any key decisions from the discussion in the PR description.
+
+### 3. Large Feature or Multi-PR Implementation
+
+If the PR introduces large-scale changes, affecting multiple areas of the codebase or requiring step-by-step integration:
+
+- Document the feature first before opening any PR.
+- Open a Github issue with a detailed technical proposal explaining the feature. This ideally will include some details like:
+  - Big picture explanation of how and why the feature will help or will change the codebase
+  - Rough outline of the classes and interfaces that will be implemented
+  - If a functional implementation is used, a brief description of what each function will do, possibly with a basic function signature if it is clear what will be needed
+  - Broad overview of how data will flow and integrate with the surrounding sub-systems
+  - Rough discussion of potential performance (cpu and memory) implications
+- Share the document with the team and gather feedback before implementation.
+- Create a feature branch to showcase the entire implementation. The idea will be to get this branch deployed on a feature group to test and tune the code on a devnet.
```

Comment:
why "on a devnet"? if this change modifies code that is already live on mainnet / public testnet we should get data from those networks and test it there